src/Symfony/Component/DependencyInjection/Resource/ParametersTrait.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Resource;
+
+trait ParametersTrait
+{
+    private array $parameters = [];
+
+    /**
+     * @var array<string, mixed> $parameters
```

Comment:
isn't mixed too broad?

Reply:
Narrowed the type a bit to `array<string, array|bool|string|int|float|\UnitEnum|null>`